### PR TITLE
CSHARP-4343: Enable AWS ECS tests.

### PR DESCRIPTION
**Pull request by DmitryLukyanov**

### CODE REVIEW

File: evergreen/evergreen.yml [hunk 1]
```diff
@@ -499,23 +498,57 @@ functions:
...
+        script: |
+          ${PREPARE_SHELL}
+          if [ "${skip_ECS_auth_test}" = "true" ]; then
+             echo "This platform does not support the ECS auth test, skipping..."
+             exit 0
+          fi
+          cd ${DRIVERS_TOOLS}/.evergreen/auth_aws
+          . ./activate_venv.sh
+          echo "Project Directory: $PROJECT_DIRECTORY"
+          # SRC_DIRECTORY is workaround since EG_TOOLS expects "src" folder as a root
```
Comment (DmitryLukyanov) [inline evergreen/evergreen.yml:539]: I've tried a number of attempts to move this logic on EG tools but with no luck. So here, I copy "mongo-csharp-driver" folder with driver sources into "src" and the content (`run-mongodb-aws-ecs-test.sh`) of "evergreen" folder to ".evergreen". All this structure is expected from EG_TOOLS scripts

File: evergreen/evergreen.yml [hunk 2]
```diff
@@ -1509,6 +1548,7 @@ axes:
         display_name: "Windows 64-bit"
         variables:
           OS: "windows-64"
+          skip_ECS_auth_test: true
```
Comment (DmitryLukyanov) [inline evergreen/evergreen.yml:1551]: ECS works only on linux

File: evergreen/evergreen.yml [hunk 3]
```diff
@@ -536,7 +569,7 @@ functions:
         working_dir: mongo-csharp-driver
         script: |
           ${PREPARE_SHELL}
-          evergreen/run-mongodb-aws-test.sh
+          OS="${OS}" ASSERT_NO_URI_CREDS=true evergreen/run-mongodb-aws-test.sh
```
Comment (JamesKovacs) [inline evergreen/evergreen.yml:572]: `${OS}` shouldn't have any spaces in it and thus doesn't need to be quoted. If we do quote it, we should quote it consistently on line 595 below.

File: tests/MongoDB.Driver.Tests/Communication/Security/AwsAuthenticationTests.cs [hunk 4]
```diff
@@ -131,6 +131,14 @@ async Task Find(IMongoClient client)
                 cursor.ToList();
             }
         }
+
+        [SkippableFact]
```
Comment (JamesKovacs) [inline tests/MongoDB.Driver.Tests/Communication/Security/AwsAuthenticationTests.cs:135]: If this is a `SkippableFact`, shouldn't we have some `RequireEnvironment.Check()` or similar statement to start?

**Review by JamesKovacs — CHANGES_REQUESTED** [review]
Overall looking good. Minor feedback. See below.

File: tests/MongoDB.Driver.Tests/Communication/Security/AwsAuthenticationTests.cs (hunk 4 above)
Comment (DmitryLukyanov) [reply tests/MongoDB.Driver.Tests/Communication/Security/AwsAuthenticationTests.cs:135]: it was skippable, but not now. Yes, it should be changed

File: evergreen/evergreen.yml (hunk 3 above)
Comment (DmitryLukyanov) [reply evergreen/evergreen.yml]: done

File: tests/MongoDB.Driver.Tests/Communication/Security/AwsAuthenticationTests.cs [hunk 5]
```diff
@@ -131,6 +131,14 @@ async Task Find(IMongoClient client)
                 cursor.ToList();
             }
         }
+
+        [Fact]
+        public void Ecs_should_fill_AWS_CONTAINER_CREDENTIALS_RELATIVE_URI()
+        {
+            var isEcs = Environment.GetEnvironmentVariable("AWS_ECS_TEST") != null;
+            var awsContainerRelativeUri = Environment.GetEnvironmentVariable("AWS_CONTAINER_CREDENTIALS_RELATIVE_URI");
+            (awsContainerRelativeUri != null).Should().Be(isEcs);
```
Comment (DmitryLukyanov) [inline tests/MongoDB.Driver.Tests/Communication/Security/AwsAuthenticationTests.cs:140]: this test was wrong, fixed

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Pull request merged**